control-plane/connect-inject/terminatinggatewayservice_controller.go

```diff
@@ -0,0 +1,440 @@
+		return ctrl.Result{}, err
+	}
+
+	spec := terminatingGatewayService.Spec
+
+	err = r.createOrUpdateService(terminatingGatewayService, ctx)
+	if err != nil {
+		r.Log.Error(err, "Unable to create or update service", "name", req.Name, "ns", req.Namespace)
+		r.updateStatusError(ctx, terminatingGatewayService, err)
+	}
+
+	// The DeletionTimestamp is zero when the object has not been marked for deletion. The finalizer is added
+	// in case it does not exist to all resources. If the DeletionTimestamp is non-zero, the object has been
+	// marked for deletion and goes into the deletion workflow.
+	if terminatingGatewayService.GetDeletionTimestamp().IsZero() {
+		if !controllerutil.ContainsFinalizer(terminatingGatewayService, FinalizerName) {
+			controllerutil.AddFinalizer(terminatingGatewayService, FinalizerName)
+			if err := r.Update(ctx, terminatingGatewayService); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		if containsString(terminatingGatewayService.Finalizers, FinalizerName) {
+			r.Log.Info("TerminatingGatewayService was deleted, deleting from consul", "name", req.Name, "ns", req.Namespace)
+			err := r.deleteService(spec.Service.ServiceName)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			controllerutil.RemoveFinalizer(terminatingGatewayService, FinalizerName)
+			err = r.Update(ctx, terminatingGatewayService)
+			return ctrl.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, err
+}
+
+func terminatingGatewayACLRole(aclRoleList []*api.ACLRole) (*api.ACLRole, error) {
+	strToFind := "terminating-gateway"
+
+	result := &api.ACLRole{}
+	roleFound := false
+
+	for _, role := range aclRoleList {
+		if strings.Contains(role.Name, strToFind) {
+			result = role
+			roleFound = true
+			break
+		}
+	}
+
+	if !roleFound {
+		return result, errors.New("Terminating Gateway ACL Role not found")
+	}
+	return result, nil
+}
+
+// updateStatus updates the terminatingGatewayService's ReconcileError in the status.
+func (r *TerminatingGatewayServiceController) updateStatus(ctx context.Context, terminatingGatewayService *consulv1alpha1.TerminatingGatewayService) error {
+
+	policyName := ""
+	if r.AclEnabled {
+		policyName = terminatingGatewayService.Spec.Service.ServiceName + "-write-policy"
+	}
+
+	terminatingGatewayService.Status.LastSyncedTime = &metav1.Time{Time: time.Now()}
+	terminatingGatewayService.SetSyncedCondition(corev1.ConditionTrue, "", "")
+
+	terminatingGatewayService.Status.ServiceInfoRef = &consulv1alpha1.ServiceInfoRefStatus{
+		ServiceName: terminatingGatewayService.Spec.Service.ServiceName,
+		PolicyName:  policyName,
+	}
+
+	err := r.Status().Update(ctx, terminatingGatewayService)
+	if err != nil {
+		r.Log.Error(err, "failed to update TerminatingGatewayService status", "name", terminatingGatewayService.Name, terminatingGatewayService.Namespace)
+	}
+	return err
+}
+
+// updateStatusError updates the terminatingGatewayService's ReconcileError in the status.
```

Comment:
The comment can be updated to say you update the Condition in the status.

Reply:
Done